common/environment/environment.cpp

```diff
@@ -1821,9 +1791,45 @@ CConstDropZoneServerInfoIterator::CConstDropZoneServerInfoIterator(const IConstD
 {
     Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
     constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
-    Owned<IPropertyTree> pSrc = &dropZone->getPTree();
-    serverListIt.setown(pSrc->getElements("ServerList"));
-    maxIndex = pSrc->getCount("ServerList");
+
+    // For backward compatibility
+    SCMStringBuffer dropZoneMachineName;
+    // Returns dropzone '@computer' value if it is exists
+    dropZone->getComputerName(dropZoneMachineName);
+
+    if (0 != dropZoneMachineName.length())
+    {
+        // Create a ServerList for legacy element.
+        legacyServerList.setown(createPTree());
+
+        Owned<IConstMachineInfo> machineInfo = constEnv->getMachine(dropZoneMachineName.str());
+        if (machineInfo)
+        {
+            SCMStringBuffer dropZoneMachineNetAddress;
+            machineInfo->getNetAddress(dropZoneMachineNetAddress);
+
+            // Create a single ServerList record related to @computer
+            //<ServerList name="ServerList" server="<IP_of_@computer>"/>
+            Owned<IPropertyTree> newRecord = createPTree();
+            newRecord->setProp("@name", "ServerList");
```

Comment:
I kept "ServerList" as name because actually I don't know who, where and how to plan to use the content of the name property, I tried to make backward compatibility transparent and maybe it can cause trouble in the future if the name is not "ServerListN".
However, if you wish to change I can do that, but in this case I think the related documentation should change as well to emphasize the difference.